### ISSUE TITLE
Invalid Activity URL

### ISSUE THREAD

**Issue by Laith98Dev** [JaxkDev/DiscordBot#84]
It looks like someone using custom status with http twitch link which crash the plugin 

I am using DiscordBot version 2.1.6

```log
[2023-01-30T13:15:44.497794+00:00] DiscordThread.DEBUG: TLS cafile set to '/root/arab/plugin_data/DiscordBot/cacert.pem' [] []
[2023-01-30T13:15:44.531488+00:00] DiscordThread.DEBUG: BUCKET getoauth2/applications/@me queued REQ GET oauth2/applications/@me [] []
[2023-01-30T13:15:44.538539+00:00] DiscordThread.DEBUG: BUCKET getgateway/bot queued REQ GET gateway/bot [] []
[2023-01-30T13:15:44.595858+00:00] DiscordThread.DEBUG: [GC] Claimed 500kb and 0 cycles. [] []
[2023-01-30T13:15:44.750207+00:00] DiscordThread.DEBUG: REQ GET gateway/bot successful [] []
[2023-01-30T13:15:44.750260+00:00] DiscordThread.DEBUG: http not checking {"waiting":1,"empty":true} []
[2023-01-30T13:15:44.750297+00:00] DiscordThread.DEBUG: BUCKET getgateway/bot queue empty [] []
[2023-01-30T13:15:44.750361+00:00] DiscordThread.INFO: gateway retrieved and set {"gateway":"wss://gateway.discord.gg/?v=8&encoding=json","session":{"total":1000,"remaining":992,"reset_after":37222660,"max_concurrency":1}} []
[2023-01-30T13:15:44.750387+00:00] DiscordThread.INFO: starting connection to websocket {"gateway":"wss://gateway.discord.gg/?v=8&encoding=json"} []
[2023-01-30T13:15:44.766705+00:00] DiscordThread.DEBUG: REQ GET oauth2/applications/@me successful [] []
[2023-01-30T13:15:44.766762+00:00] DiscordThread.DEBUG: http not checking {"waiting":0,"empty":true} []
[2023-01-30T13:15:44.766792+00:00] DiscordThread.DEBUG: BUCKET getoauth2/applications/@me queue empty [] []
[2023-01-30T13:15:44.958649+00:00] DiscordThread.INFO: websocket connection has been created [] []
[2023-01-30T13:15:44.959385+00:00] DiscordThread.INFO: received hello [] []
[2023-01-30T13:15:44.959416+00:00] DiscordThread.DEBUG: sending heartbeat {"seq":null} []
[2023-01-30T13:15:44.959468+00:00] DiscordThread.INFO: heartbeat timer initilized {"interval":41250.0} []
[2023-01-30T13:15:44.959495+00:00] DiscordThread.INFO: identifying {"payload":{"op":2,"d":{"token":"xxxxxx","properties":{"$os":"Linux","$browser":"DiscordBot (https://github.com/discord-php/DiscordPHP-HTTP, v8.1.2)","$device":"DiscordBot (https://github.com/discord-php/DiscordPHP-HTTP, v8.1.2)","$referrer":"https://github.com/discord-php/DiscordPHP","$referring_domain":"https://github.com/discord-php/DiscordPHP"},"compress":true,"intents":32767}}} []
[2023-01-30T13:15:45.063182+00:00] DiscordThread.DEBUG: received heartbeat ack {"response_time":103.6989688873291} []
[2023-01-30T13:15:45.201258+00:00] DiscordThread.DEBUG: ready packet received [] []
[2023-01-30T13:15:45.201300+00:00] DiscordThread.DEBUG: discord trace received {"trace":["[\"gateway-prd-us-east1-b-xghg\",{\"micros\":141940,\"calls\":[\"id_created\",{\"micros\":2150,\"calls\":[]},\"session_lookup_time\",{\"micros\":313,\"calls\":[]},\"session_lookup_finished\",{\"micros\":16,\"calls\":[]},\"discord-sessions-blue-prd-2-280\",{\"micros\":138045,\"calls\":[\"start_session\",{\"micros\":71123,\"calls\":[\"discord-api-8574dfcf9b-tzhk6\",{\"micros\":62913,\"calls\":[\"get_user\",{\"micros\":12087},\"get_guilds\",{\"micros\":8177},\"send_scheduled_deletion_message\",{\"micros\":10},\"guild_join_requests\",{\"micros\":1061},\"authorized_ip_coro\",{\"micros\":12}]}]},\"starting_guild_connect\",{\"micros\":60,\"calls\":[]},\"presence_started\",{\"micros\":299,\"calls\":[]},\"guilds_started\",{\"micros\":153,\"calls\":[]},\"guilds_connect\",{\"micros\":2,\"calls\":[]},\"presence_connect\",{\"micros\":66380,\"calls\":[]},\"connect_finished\",{\"micros\":66386,\"calls\":[]},\"build_ready\",{\"micros\":20,\"calls\":[]},\"optimize_ready\",{\"micros\":1,\"calls\":[]},\"split_ready\",{\"micros\":1,\"calls\":[]},\"clean_ready\",{\"micros\":0,\"calls\":[]}]}]}]"]} []
[2023-01-30T13:15:45.201873+00:00] DiscordThread.DEBUG: client created and session id stored {"session_id":"6877eb076d07fa3465b01696e1d0d1fe","user":{"id":"926835771548241931","username":"ARAB SYSTEM","avatar":"https://cdn.discordapp.com/avatars/926835771548241931/f7548dc94661d829d83b949e629a4297.jpg?size=1024","discriminator":"6981","bot":true,"system":null,"mfa_enabled":null,"locale":null,"verified":true,"email":null,"flags":null,"premium_type":null,"public_flags":null}} []
[2023-01-30T13:15:45.201906+00:00] DiscordThread.INFO: did not parse private channels [] []
[2023-01-30T13:15:45.203629+00:00] DiscordThread.INFO: stored guilds {"count":0,"unavailable":2} []
[2023-01-30T13:15:45.266570+00:00] DiscordThread.DEBUG: guild available {"guild":"854689029852102707","unavailable":2} []
[2023-01-30T13:15:45.442769+00:00] DiscordThread.DEBUG: guild available {"guild":"866998700314984458","unavailable":1} []
[2023-01-30T13:15:45.442823+00:00] DiscordThread.INFO: all guilds are now available {"count":2} []
[2023-01-30T13:15:45.442857+00:00] DiscordThread.INFO: set up chunking, checking for chunks every 5 seconds [] []
[2023-01-30T13:15:45.442877+00:00] DiscordThread.DEBUG: sending 1 chunks with 2 large guilds overall [] []
[2023-01-30T13:15:45.442895+00:00] DiscordThread.DEBUG: sending chunk with 2 large guilds [] []
[2023-01-30T13:15:45.567450+00:00] DiscordThread.DEBUG: received guild member chunk {"guild_id":"854689029852102707","guild_name":"Laith98Dev Server","chunk_count":577,"member_collection":577,"member_count":577} []
[2023-01-30T13:15:45.567914+00:00] DiscordThread.DEBUG: parsed 0 members (skipped 577) {"repository_count":577,"actual_count":577} []
[2023-01-30T13:15:45.567967+00:00] DiscordThread.DEBUG: all users have been loaded {"guild":"854689029852102707","member_collection":577,"member_count":577} []
[2023-01-30T13:15:46.084097+00:00] DiscordThread.DEBUG: received guild member chunk {"guild_id":"866998700314984458","guild_name":"ArabSkills Community Official","chunk_count":1000,"member_collection":2513,"member_count":2513} []
[2023-01-30T13:15:46.085027+00:00] DiscordThread.DEBUG: parsed 0 members (skipped 1000) {"repository_count":2513,"actual_count":2513} []
[2023-01-30T13:15:46.085063+00:00] DiscordThread.DEBUG: all users have been loaded {"guild":"866998700314984458","member_collection":2513,"member_count":2513} []
[2023-01-30T13:15:46.085090+00:00] DiscordThread.INFO: client is ready [] []
[2023-01-30T13:15:46.085766+00:00] DiscordThread.DEBUG: Starting the data pack, please be patient. [] []
[2023-01-30T13:15:46.088686+00:00] DiscordThread.DEBUG: BUCKET getguilds/854689029852102707/bans queued REQ GET guilds/854689029852102707/bans [] []
[2023-01-30T13:15:46.096544+00:00] DiscordThread.DEBUG: BUCKET getguilds/854689029852102707/invites queued REQ GET guilds/854689029852102707/invites [] []
[2023-01-30T13:15:46.246466+00:00] DiscordThread.DEBUG: BUCKET getguilds/866998700314984458/bans queued REQ GET guilds/866998700314984458/bans [] []
[2023-01-30T13:15:46.252239+00:00] DiscordThread.DEBUG: BUCKET getguilds/866998700314984458/invites queued REQ GET guilds/866998700314984458/invites [] []
[2023-01-30T13:15:46.656293+00:00] DiscordThread.DEBUG: Data pack took: 0.57045s & 6mb of memory, Final size: 6412 [] []
[2023-01-30T13:15:46.664819+00:00] DiscordThread.INFO: Client 'ARAB SYSTEM#6981' ready. [] []
[2023-01-30T13:15:46.706999+00:00] DiscordThread.DEBUG: received guild member chunk {"guild_id":"866998700314984458","guild_name":"ArabSkills Community Official","chunk_count":1000,"member_collection":2513,"member_count":2513} []
[2023-01-30T13:15:46.707913+00:00] DiscordThread.DEBUG: parsed 0 members (skipped 1000) {"repository_count":2513,"actual_count":2513} []
[2023-01-30T13:15:46.707956+00:00] DiscordThread.DEBUG: all users have been loaded {"guild":"866998700314984458","member_collection":2513,"member_count":2513} []
[2023-01-30T13:15:46.716034+00:00] DiscordThread.DEBUG: received guild member chunk {"guild_id":"866998700314984458","guild_name":"ArabSkills Community Official","chunk_count":513,"member_collection":2513,"member_count":2513} []
[2023-01-30T13:15:46.716405+00:00] DiscordThread.DEBUG: parsed 0 members (skipped 513) {"repository_count":2513,"actual_count":2513} []
[2023-01-30T13:15:46.716444+00:00] DiscordThread.DEBUG: all users have been loaded {"guild":"866998700314984458","member_collection":2513,"member_count":2513} []
[2023-01-30T13:15:46.979747+00:00] DiscordThread.DEBUG: REQ GET guilds/854689029852102707/invites successful [] []
[2023-01-30T13:15:46.979818+00:00] DiscordThread.DEBUG: http not checking {"waiting":3,"empty":true} []
[2023-01-30T13:15:46.979857+00:00] DiscordThread.DEBUG: BUCKET getguilds/854689029852102707/invites queue empty [] []
[2023-01-30T13:15:46.981050+00:00] DiscordThread.DEBUG: Successfully fetched 27 invites from server 'Laith98Dev Server' (854689029852102707) [] []
[2023-01-30T13:15:47.056967+00:00] DiscordThread.DEBUG: REQ GET guilds/854689029852102707/bans successful [] []
[2023-01-30T13:15:47.057021+00:00] DiscordThread.DEBUG: http not checking {"waiting":2,"empty":true} []
[2023-01-30T13:15:47.057054+00:00] DiscordThread.DEBUG: BUCKET getguilds/854689029852102707/bans queue empty [] []
[2023-01-30T13:15:47.057572+00:00] DiscordThread.DEBUG: Successfully fetched 13 bans from server 'Laith98Dev Server' (854689029852102707) [] []
[2023-01-30T13:15:47.058780+00:00] DiscordThread.DEBUG: REQ GET guilds/866998700314984458/bans successful [] []
[2023-01-30T13:15:47.058804+00:00] DiscordThread.DEBUG: http not checking {"waiting":1,"empty":true} []
[2023-01-30T13:15:47.058824+00:00] DiscordThread.DEBUG: BUCKET getguilds/866998700314984458/bans queue empty [] []
[2023-01-30T13:15:47.059193+00:00] DiscordThread.DEBUG: Successfully fetched 68 bans from server 'ArabSkills Community Official' (866998700314984458) [] []
[2023-01-30T13:15:47.219171+00:00] DiscordThread.DEBUG: REQ GET guilds/866998700314984458/invites successful [] []
[2023-01-30T13:15:47.219244+00:00] DiscordThread.DEBUG: http not checking {"waiting":0,"empty":true} []
[2023-01-30T13:15:47.219292+00:00] DiscordThread.DEBUG: BUCKET getguilds/866998700314984458/invites queue empty [] []
[2023-01-30T13:15:47.224656+00:00] DiscordThread.DEBUG: Successfully fetched 209 invites from server 'ArabSkills Community Official' (866998700314984458) [] []
[2023-01-30T13:15:47.979024+00:00] DiscordThread.DEBUG: BUCKET getguilds/854689029852102707/invites queue empty [] []
[2023-01-30T13:15:48.218450+00:00] DiscordThread.DEBUG: BUCKET getguilds/866998700314984458/invites queue empty [] []
[2023-01-30T13:15:49.750445+00:00] DiscordThread.DEBUG: BUCKET getgateway/bot queue empty [] []
[2023-01-30T13:15:57.057219+00:00] DiscordThread.DEBUG: BUCKET getguilds/854689029852102707/bans queue empty [] []
[2023-01-30T13:15:57.059034+00:00] DiscordThread.DEBUG: BUCKET getguilds/866998700314984458/bans queue empty [] []
[2023-01-30T13:16:26.209540+00:00] DiscordThread.DEBUG: sending heartbeat {"seq":46} []
[2023-01-30T13:16:26.312597+00:00] DiscordThread.DEBUG: received heartbeat ack {"response_time":102.9210090637207} []
[2023-01-30T13:16:26.709881+00:00] DiscordThread.CRITICAL: AssertionError: "Invalid url 'http://twitch.tv/doorsteel119'." (EXCEPTION) in "phar:///root/arab/plugins/DiscordBot.phar/src/Models/Activity.php" at line 169 [] []
[2023-01-30T13:16:26.710160+00:00] DiscordThread.CRITICAL: #0 /root/arab/plugins/DiscordBot.phar/src/Models/Activity(104): JaxkDev\DiscordBot\Models\Activity->setUrl(string[29] http://twitch.tv/doorsteel119) [] []
[2023-01-30T13:16:26.710194+00:00] DiscordThread.CRITICAL: #1 /root/arab/plugins/DiscordBot.phar/src/Bot/ModelConverter(93): JaxkDev\DiscordBot\Models\Activity->__construct(string[12] InamS On Top, integer 1, integer 1675084586591, string[29] http://twitch.tv/doorsteel119, NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL , NULL ) [] []
[2023-01-30T13:16:26.710214+00:00] DiscordThread.CRITICAL: #2 /root/arab/plugins/DiscordBot.phar/src/Bot/Handlers/DiscordEventHandler(324): JaxkDev\DiscordBot\Bot\ModelConverter::genModelActivity(object Discord\Parts\User\Activity#17013) [] []
[2023-01-30T13:16:26.710232+00:00] DiscordThread.CRITICAL: #3 /root/arab/plugins/DiscordBot.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait(123): JaxkDev\DiscordBot\Bot\Handlers\DiscordEventHandler->onPresenceUpdate(object Discord\Parts\WebSockets\PresenceUpdate#9033, object JaxkDev\DiscordBot\Bot\Discord#12, object Discord\Parts\WebSockets\PresenceUpdate#6906) [] []
[2023-01-30T13:16:26.710250+00:00] DiscordThread.CRITICAL: #4 /root/arab/plugins/DiscordBot.phar/vendor/team-reflex/discord-php/src/Discord/Discord(697): Discord\Discord->emit(string[15] PRESENCE_UPDATE, array[3]) [] []
[2023-01-30T13:16:26.710282+00:00] DiscordThread.CRITICAL: #5 /root/arab/plugins/DiscordBot.phar/vendor/react/promise/src/FulfilledPromise(42): Discord\Discord->Discord\{closure}(array[2]) [] []
[2023-01-30T13:16:26.710300+00:00] DiscordThread.CRITICAL: #6 /root/arab/plugins/DiscordBot.phar/vendor/react/promise/src/Promise(66): React\Promise\FulfilledPromise->done(object Closure#9165, object Closure#6425) [] []
[2023-01-30T13:16:26.710343+00:00] DiscordThread.CRITICAL: #7 /root/arab/plugins/DiscordBot.phar/vendor/react/promise/src/Promise(168): React\Promise\Promise::React\Promise\{closure}(object React\Promise\FulfilledPromise#9039) [] []
[2023-01-30T13:16:26.710360+00:00] DiscordThread.CRITICAL: #8 /root/arab/plugins/DiscordBot.phar/vendor/react/promise/src/Promise(231): React\Promise\Promise->settle(object React\Promise\FulfilledPromise#9039) [] []
[2023-01-30T13:16:26.710376+00:00] DiscordThread.CRITICAL: #9 /root/arab/plugins/DiscordBot.phar/vendor/react/promise/src/Deferred(36): React\Promise\Promise::React\Promise\{closure}(array[2]) [] []
[2023-01-30T13:16:26.710392+00:00] DiscordThread.CRITICAL: #10 /root/arab/plugins/DiscordBot.phar/vendor/team-reflex/discord-php/src/Discord/WebSockets/Events/PresenceUpdate(37): React\Promise\Deferred->resolve(array[2]) [] []
[2023-01-30T13:16:26.710408+00:00] DiscordThread.CRITICAL: #11 /root/arab/plugins/DiscordBot.phar/vendor/team-reflex/discord-php/src/Discord/Discord(721): Discord\WebSockets\Events\PresenceUpdate->handle(object Discord\Helpers\Deferred#1036, object stdClass#9468) [] []
[2023-01-30T13:16:26.710424+00:00] DiscordThread.CRITICAL: #12 /root/arab/plugins/DiscordBot.phar/vendor/team-reflex/discord-php/src/Discord/Discord(594): Discord\Discord->handleDispatch(object stdClass#9045) [] []
[2023-01-30T13:16:26.710440+00:00] DiscordThread.CRITICAL: #13 /root/arab/plugins/DiscordBot.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait(123): Discord\Discord->handleWsMessage(object Ratchet\RFC6455\Messaging\Message#9307, object Ratchet\Client\WebSocket#119) [] []
[2023-01-30T13:16:26.710456+00:00] DiscordThread.CRITICAL: #14 /root/arab/plugins/DiscordBot.phar/vendor/ratchet/pawl/src/WebSocket(72): Ratchet\Client\WebSocket->emit(string[7] message, array[2]) [] []
[2023-01-30T13:16:26.710472+00:00] DiscordThread.CRITICAL: #15 /root/arab/plugins/DiscordBot.phar/vendor/ratchet/rfc6455/src/Messaging/MessageBuffer(248): Ratchet\Client\WebSocket->Ratchet\Client\{closure}(object Ratchet\RFC6455\Messaging\Message#9307, object Ratchet\RFC6455\Messaging\MessageBuffer#225) [] []
[2023-01-30T13:16:26.710488+00:00] DiscordThread.CRITICAL: #16 /root/arab/plugins/DiscordBot.phar/vendor/ratchet/rfc6455/src/Messaging/MessageBuffer(194): Ratchet\RFC6455\Messaging\MessageBuffer->processData(string[562] .~..{"t":"PRESENCE_UPDATE","s":47,"op":0,"d":{"user":{"id":"1013020384129462292") [] []
[2023-01-30T13:16:26.710504+00:00] DiscordThread.CRITICAL: #17 /root/arab/plugins/DiscordBot.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait(123): Ratchet\RFC6455\Messaging\MessageBuffer->onData(string[562] .~..{"t":"PRESENCE_UPDATE","s":47,"op":0,"d":{"user":{"id":"1013020384129462292") [] []
[2023-01-30T13:16:26.710520+00:00] DiscordThread.CRITICAL: #18 /root/arab/plugins/DiscordBot.phar/vendor/react/stream/src/Util(71): Evenement\EventEmitter->emit(string[4] data, array[1]) [] []
[2023-01-30T13:16:26.710536+00:00] DiscordThread.CRITICAL: #19 /root/arab/plugins/DiscordBot.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait(123): React\Stream\Util::React\Stream\{closure}(string[562] .~..{"t":"PRESENCE_UPDATE","s":47,"op":0,"d":{"user":{"id":"1013020384129462292") [] []
[2023-01-30T13:16:26.710552+00:00] DiscordThread.CRITICAL: #20 /root/arab/plugins/DiscordBot.phar/vendor/react/stream/src/DuplexResourceStream(196): Evenement\EventEmitter->emit(string[4] data, array[1]) [] []
[2023-01-30T13:16:26.710573+00:00] DiscordThread.CRITICAL: #21 /root/arab/plugins/DiscordBot.phar/vendor/react/event-loop/src/StreamSelectLoop(246): React\Stream\DuplexResourceStream->handleData(resource Resource id #327) [] []
[2023-01-30T13:16:26.710589+00:00] DiscordThread.CRITICAL: #22 /root/arab/plugins/DiscordBot.phar/vendor/react/event-loop/src/StreamSelectLoop(213): React\EventLoop\StreamSelectLoop->waitForStreamActivity(integer 50000) [] []
[2023-01-30T13:16:26.710605+00:00] DiscordThread.CRITICAL: #23 /root/arab/plugins/DiscordBot.phar/vendor/team-reflex/discord-php/src/Discord/Discord(1368): React\EventLoop\StreamSelectLoop->run() [] []
[2023-01-30T13:16:26.710621+00:00] DiscordThread.CRITICAL: #24 /root/arab/plugins/DiscordBot.phar/src/Bot/Client(111): Discord\Discord->run() [] []
[2023-01-30T13:16:26.710637+00:00] DiscordThread.CRITICAL: #25 /root/arab/plugins/DiscordBot.phar/src/Communication/BotThread(51): JaxkDev\DiscordBot\Bot\Client->__construct(object JaxkDev\DiscordBot\Communication\BotThread#1, array[4]) [] []
[2023-01-30T13:16:26.710654+00:00] DiscordThread.CRITICAL: #26 (): JaxkDev\DiscordBot\Communication\BotThread->run() [] []
[2023-01-30T13:16:26.710740+00:00] DiscordThread.INFO: discord closed [] []
[2023-01-30T13:16:26.710766+00:00] DiscordThread.NOTICE: Discord thread closed. [] []
```

**Comment by JaxkDev** [comment]
URL has been nothing but issues with what I thought originally was a strict format but apparently not.

Will be fixed in next update.

**Comment by Laith98Dev** [comment]
Okay

**Comment by JaxkDev** [comment]
Apologies, fixed in v2.1.8 (e6390ba)
